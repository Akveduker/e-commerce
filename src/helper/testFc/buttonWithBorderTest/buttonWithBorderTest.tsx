import { render, screen } from "@testing-library/react"
import { ButtonWithBorderProps } from "../../../types/ui/buttonsWithBorders/buttonsWithBorders"
import { MemoryRouter } from "react-router-dom";

export const buttonWithBorderTest = (name: string, ButtonElement: React.ReactElement<ButtonWithBorderProps>, childrenText: string) => {
    describe(name, () => {
        test('Снапшот', () => {
            const { container } = render(
                <MemoryRouter>
                    {ButtonElement}
                </MemoryRouter>
            )
            expect(container).toMatchSnapshot()
        })
        test('чилдрен есть', () => {
            render(
                <MemoryRouter>
                    {ButtonElement}
                </MemoryRouter>
            )
            const button = screen.getByText(childrenText)
            expect(button).toHaveTextContent(childrenText)
        })
        test('Классы работают', () => {
            render(
                <MemoryRouter>
                    {ButtonElement}
                </MemoryRouter>
            )
            const button = screen.getByText(childrenText)
            expect(button).toHaveClass(`button--${ButtonElement.props.styleType}`)

        })
    })
}