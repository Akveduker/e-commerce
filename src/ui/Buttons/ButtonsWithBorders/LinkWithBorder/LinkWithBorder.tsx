import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import { LinkWithBorderProps } from '../../../../types/ui/buttonsWithBorders/buttonsWithBorders';

import s from '../buttonsStyles.module.scss'
const LinkWithBorder: FC<LinkWithBorderProps> = ({ styleType, children, ...props }) => {
    const className = `${s['button']} ${s[`button--${styleType}`]}`
    return (
        <Link {...props} className={className}>
            {children}
        </Link>
    );
};

export default LinkWithBorder;