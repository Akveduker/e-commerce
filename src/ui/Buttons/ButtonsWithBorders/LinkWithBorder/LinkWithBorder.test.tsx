import { buttonWithBorderTest } from "../../../../helper/testFc/buttonWithBorderTest/buttonWithBorderTest";
import LinkWithBorder from "./LinkWithBorder";

buttonWithBorderTest(
    'buttonWithBorderTest',
    <LinkWithBorder
        styleType="medium"
        to={'/'}
    >
        test
    </LinkWithBorder>,
    'test'
)