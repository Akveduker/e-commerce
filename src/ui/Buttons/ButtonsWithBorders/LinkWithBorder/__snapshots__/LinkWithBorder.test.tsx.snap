// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonWithBorderTest Снапшот 1`] = `
<div>
  <a
    class="button button--medium"
    href="/"
  >
    test
  </a>
</div>
`;
